### DISCUSSION TITLE
Full blown TLS/SSL for WebSend in ESL32?

### DISCUSSION THREAD

**Discussion by hagenbuch** [arendst/Tasmota#19724] [Q&A] [answered]
Hi all,

I understand why TLS/SSL is hardly possible for a chip with the limitations of the ESP8266 - or only in a limited way with manually installed certificates.

Now, when searching for TLS here, I find only references in combination with MQTT.

Is "fluu blown" TLS on ESP32 already useable for WebSend etc.? My devices would have to get or send data via https and it would be fantastic if either the complete TLS trust chain could be checked or else if one could add just the certificate authorities we'd need most - I guess that would be letsencrypt..

If there be not enough flash memory to host all the certs it would be nice to at least offer that option if a SDcard is present (which is true fpr all my projects).

Andreas

**Comment by s-hadinger** [comment, marked as answer]
There is no plan for certificate validation with https. It's a can of worms to do it right. Currently any certificate is accepted when doing an https request

**Comment by hagenbuch** [comment]
Well that's already something, thanks! To put only Letsencrypt's CA in the device, would that be a smaller can? :)

**Comment by Jason2866** [comment]
Feel free to do :-)
